### DISCUSSION TITLE
Transmission says port is closed but it is verifiably open

### DISCUSSION THREAD

**Discussion by realbaylen** [transmission/transmission#8433] [Support]
### What is the issue?

My port is verifiably open (checked with https://www.yougetsignal.com/tools/open-ports/), but Transmission says it is is closed (port 51413). I am not sure if Transmission is acting like it is open or not... on the Arch Linux ISO, it says it is seeding to 0 of 40 peers. Maybe this means it is acting like it is actually closed?

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

Transmission 4.0.5

**Comment by asavageiv** [comment]
I'm seeing the same issue. Using the Network tab in settings on the web interface I see the port is closed, but https://www.canyouseeme.org/ shows it as open and tcpdump shows traffic. It seems like seeding is working.

Is there any further clarification or testing I can do?

Transmission 4.0.6

**Comment by kousu** [comment]
I set up a port forward in my router and was able to connect over it using `nc` (TCP) on both ends. With transmission running I tried canyouseeme like suggested in [the docs](https://github.com/transmission/transmission/blob/64268a3a7f14f18db43184812dad91462e8604d5/docs/Why-is-my-port-closed.md) and it says I'm fine. With `nc -l -p 51413` running locally (with Tranmission temporarily moved to another port) I can see peers connecting to me. With transmission back on port 51413 (and no server-side `nc`) I can connect over `nc` to that port; and if I move transmission to port 51412 I can no longer connect. If I `nc -u -l -p 51413` even while Transmission is running I can see data coming in -- mostly metadata requests from peers. So the port forward is working in every way I can tell. But the "Test Port" button always reports **closed**.

Using tcpdump I can see packets going to canyouseeme when I click its test button, but not when I click Transmission's test button; so what does Transmission *actually* use to test if a port is open? A useful goal one of us could take on is digging in the code to find the test and reproduce it in the docs.

I have NOT forwarded IPv6, if that's relevant, only IPv4. I shouldn't have to. My router doesn't seem to give me the option, anyway.

My router doesn't use NAT-PMP or UPnP, if that's relevant, and I have that option disabled in the Transmission prefs.

I'm running transmission-gtk 4.0.6.

**Comment by kousu** [comment]
It occurs to me that @realbaylen was on Arch; did it ever work for you in the end? I'm also on Arch. @asavageiv are you on Arch?

If that's the common thread, well, [these](https://gitlab.archlinux.org/archlinux/packaging/packages/transmission/-/blob/995252779da324adcaa75072257f125cc7623689/PKGBUILD?ref_type=heads#L44) are Arch's build flags, is there a clue in there:

```

	cmake \
		-G Ninja \
		-D CMAKE_BUILD_TYPE=RelWithDebInfo \
		-D CMAKE_INSTALL_PREFIX=/usr \
		-D ENABLE_CLI=ON \
		-D ENABLE_DAEMON=ON \
		-D ENABLE_GTK=ON \
		-D ENABLE_MAC=OFF \
		-D ENABLE_QT=ON \
		-D REBUILD_WEB=ON \
		-D ENABLE_TESTS=ON \
		-D ENABLE_UTILS=ON \
		-D ENABLE_UTP=ON \
		-D INSTALL_LIB=ON \
		-D USE_SYSTEM_B64=ON \
		-D USE_SYSTEM_DEFLATE=ON \
		-D USE_SYSTEM_DHT=ON \
		-D USE_SYSTEM_EVENT2=ON \
		-D USE_SYSTEM_MINIUPNPC=ON \
		-D USE_SYSTEM_NATPMP=ON \
		-D USE_SYSTEM_PSL=ON \
		-D USE_SYSTEM_UTP=OFF \
		-D WITH_CRYPTO=openssl \
		-S . \
		-B build
```

and these are the deps:

```
	depends=(curl
	         gtk4
	         gtkmm-4.0
	         hicolor-icon-theme
	         libb64
	         libdeflate
	         libevent
	         libnatpmp
	         miniupnpc)
```

I can't remember the last time I ever saw the port checker work.

**Reply by asavageiv** [reply]
I'm in a docker container on a debian server. IDK what the container is running. Probably alpine. I'm also no longer having this issue in 4.1.0
